packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -299,6 +319,54 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
+  ///     they are considered a batch, the gesture will update by the rules above.
+  ///   Example:
+  ///     When two pointers (p1 and p2) down, the sequence of [PointerMoveEvent]s is : p1 down, p2 down, p1 move +50, p2 move +10,
+  ///     p1 move +30, p2 move -10 , p1 up and p2 up.
+  ///     The [handleEvent] process will be:
+  ///       p1 down ([_multiPointerStartTrackers].add),
+  ///       p2 down([_multiPointerStartTrackers].add),
+  ///       p1 move +50 ([_multiPointerMoveTrackers].add and the gesture wait),
+  ///       p2 move +10 ([_multiPointerMoveTrackers].add, the gesture update +50, and [_multiPointerMoveTrackers].clear),
+  ///       p1 move +30 ([_multiPointerMoveTrackers].add and the gesture wait),
+  ///       p2 move -10 ([_multiPointerMoveTrackers].add, the gesture update +20, and [_multiPointerMoveTrackers].clear),
+  ///       p1 up ([_multiPointerStartTrackers].clear),
+  ///       p2 up.
+  ///
+  ///   2) Only some of pointers moves (it means the pointers' [PointerMoveEvent] or [PointerPanZoomUpdateEvent]
+  ///   that are not moving do not come instead of [Offset] zero:
+  ///   The gesture will move if two [PointerMoveEvent] or [PointerPanZoomUpdateEvent]s of the same pointer come.
+  ///   The gesture will move delta(combined) = max(delta(i) which are positive) + min(delta(i) which are negative)
+  ///   The drag location will be the average of these moved pointers.
+  ///   Logic:
+  ///     When a [PointerMoveEvent] or [PointerPanZoomUpdateEvent] comes, add the [PointerEvent] in
+  ///     the [_multiPointerMoveTrackers]. When a [PointerMoveEvent] or [PointerPanZoomUpdateEvent] comes which pointer has been in
+  ///     the [_multiPointerMoveTrackers], the gesture will update immediately then add this [PointerEvent] in the [_multiPointerMoveTrackers].
+  ///   Example:
+  ///     When two pointers (p1 and p2) down, the sequence of [PointerMoveEvent]s is : p1 down, p2 down, p1 move +50, p1 move +10,
+  ///     p1 move +30, p1 move -10 , p1 up and p2 up.
+  ///     The [handleEvent] process will be:
+  ///       p1 down ([_multiPointerStartTrackers].add),
+  ///       p2 down ([_multiPointerStartTrackers].add),
+  ///       p1 move +50 ([_multiPointerMoveTrackers].add and the gesture wait),
+  ///       p1 move +10 ([_multiPointerMoveTrackers].clear, the gesture update +50, and [_multiPointerMoveTrackers].add,),
+  ///       p1 move +30 ([_multiPointerMoveTrackers].clear, the gesture update +10, and [_multiPointerMoveTrackers].add,),
+  ///       p1 move -10 ([_multiPointerMoveTrackers].clear, the gesture update +30, and [_multiPointerMoveTrackers].add,),
+  ///       p1 up ([_multiPointerStartTrackers].clear),
+  ///       p2 up.
+  ///       It means the last [PointerEvent] will be lost, this is acceptable. If the gesture update the last [PointerEvent]
+  ///       in the pointer up stage, it may cause flickering.
+  ///
+  /// When any pointer up which is accepted, the another pointers are given up.
```

Comment:
Do you mean "When any accepted pointer is up, all other pointers are given up."? But this is not how the native behavior of either iOS or Android works.

Reply:
Oh! I never noticed this behavior until just now when I tested. Great observation.

But I also found that it isn't always the case. For example, on iOS, if I scroll using pointer 1 first, then stop without lifting and keep it absolutely still, add a second pointer to scroll, then lift pointer 1, pointer 2 will still work. 

I admit that this is a very rare case since I may fail it even when I intentionally try so.

Reply:
> Oh! I never noticed this behavior until just now when I tested. Great observation.
> 
> But I also found that it isn't always the case. For example, on iOS, if I scroll using pointer 1 first, then stop without lifting and keep it absolutely still, add a second pointer to scroll, then lift pointer 1, pointer 2 will still work.
> 
> I admit that this is a very rare case since I may fail it even when I intentionally try so.

@dkwingsmt 

Yes, I test it and I found the logic of scroll on iOS. If one pointer leaves and no pointers are moving, then all pointers are given up. If one pointer leaves and another pointer is moving, then only the leaved pointer is given up. The failed situation of your trial is that you leave p1 but p2 not moving. I noticed just now, that the logic is easy to reproduce, but I do not think this confused logic is good. What about your opinions? Need I add this logic to flutter?

Reply:
in a vertical scrollview, delta1.x > delta2.x, and pointer1 leaves, should give up the pointer2?

Reply:
It should consider delta.y only in a vertical scroll view!